src/main/java/org/junitpioneer/jupiter/TempDirectory.java

```diff
@@ -286,5 +302,21 @@ private FileVisitResult deleteAndContinue(Path path) throws IOException {
 				}
 			});
 		}
+
+		static void deleteOnExit(Path path) {
+			if (path == null) {
+				return;
+			}
+			try {
+				File file = path.toFile();
+				if (file == null) {
+					return;
+				}
+				file.deleteOnExit();
```

Comment:
Hmm. I don't think we can assume that `Path`s will always be translatable to `File`s, because the user may pass in a filesystem other than the default disk-based one e.g. Jimfs. 🤔

Reply:
Oh sorry, I see that you already take care of this in the `catch` block below. Please ignore my earlier review/comment. :)